src/shared_modules/utils/tests/windowsHelper_test.cpp

```diff
@@ -62,4 +62,126 @@ TEST_F(WindowsHelperTest, ipv6NetMask_INVALID)
+                                        0x03, 0xbf, 0x80, 0x9a, 0x8b, 0x3f, 0x01, 0x00, 0x17, 0x00, 0x03, 0x0f, 0x04, 0x06, 0xff, 0xff, 0x44, 0x65, 0x6c, 0x6c, 0x20, 0x49, 0x6e, 0x63,
+                                        0x2e, 0x00, 0x41, 0x30, 0x31, 0x00, 0x30, 0x38, 0x2f, 0x31, 0x33, 0x2f, 0x32, 0x30, 0x31, 0x33, 0x00, 0x00, 0x01, 0x1b, 0x01, 0x00, 0x01, 0x02,
+                                        0x03, 0x04, 0x44, 0x45, 0x4c, 0x4c, 0x59, 0x00, 0x10, 0x44, 0x80, 0x38, 0xb6, 0xc0, 0x4f, 0x35, 0x59, 0x31, 0x06, 0x05, 0x06, 0x44, 0x65, 0x6c,
+                                        0x6c, 0x20, 0x49, 0x6e, 0x63, 0x2e, 0x00, 0x50, 0x6f, 0x77, 0x65, 0x72, 0x45, 0x64, 0x67, 0x65, 0x20, 0x54, 0x32, 0x30, 0x00, 0x30, 0x31, 0x00,
+                                        0x36, 0x59, 0x44, 0x38, 0x35, 0x59, 0x31, 0x00, 0x50, 0x6f, 0x77, 0x65, 0x72, 0x45, 0x64, 0x67, 0x65, 0x20, 0x54, 0x32, 0x30, 0x00, 0x20, 0x00,
+                                        0x00, 0x02, 0x0f, 0x02, 0x00, 0x01, 0x02, 0x03, 0x04, 0x00, 0x09, 0x00, 0x03, 0x00, 0x0a, 0x00, 0x44, 0x65, 0x6c, 0x6c, 0x20, 0x49, 0x6e, 0x63,
+                                        0x2e, 0x00, 0x30, 0x56, 0x44, 0x35, 0x48, 0x59, 0x00, 0x41, 0x30, 0x30, 0x00, 0x2f, 0x36, 0x59, 0x44, 0x38, 0x35, 0x59, 0x31, 0x2f, 0x43, 0x4e,
+                                        0x37, 0x32, 0x32, 0x30, 0x30, 0x33, 0x39, 0x33, 0x30, 0x30, 0x33, 0x39, 0x2f, 0x00, 0x20, 0x00, 0x20, 0x00, 0x00};
+
+    const auto size { rawData.size() };
+    constexpr auto SERIAL_NUMBER_DATA { "/6YD85Y1/CN722003930039/" };
+    std::string serialNumber;
+
+    PRawSMBIOSData smbios { reinterpret_cast<PRawSMBIOSData>(rawData.data()) };
+    serialNumber = Utils::parseRawSmbios(smbios->SMBIOSTableData, size);
+    EXPECT_EQ(SERIAL_NUMBER_DATA, serialNumber);
+}
+
+TEST_F(WindowsHelperTest, parserRawSMBIOSCorruptedData1_TEST)
+{
+    std::vector<unsigned char> rawData {0x0, 0x0, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0xff, 0x27, 0xc7};
+    const auto size { rawData.size() };
+    constexpr auto SERIAL_NUMBER_DATA { "" };
+    std::string serialNumber;
+
+    serialNumber = Utils::parseRawSmbios(rawData.data(), size);
+    EXPECT_EQ(SERIAL_NUMBER_DATA, serialNumber);
+}
+
+
+TEST_F(WindowsHelperTest, parserRawSMBIOSCorruptedData2_TEST)
+{
+    std::vector<unsigned char> rawData {0x2, 0x0, 0xff, 0xff, 0xff, 0xff, 0x0, 0x5, 0xa};
+    const auto size { rawData.size() };
+    constexpr auto SERIAL_NUMBER_DATA { "" };
+    std::string serialNumber;
+
+    serialNumber = Utils::parseRawSmbios(rawData.data(), size);
+    EXPECT_EQ(SERIAL_NUMBER_DATA, serialNumber);
+}
+
+TEST_F(WindowsHelperTest, parserRawSMBIOSTablesNoEndDOubleNull_TEST)
+{
+    std::vector<unsigned char> rawData {0xda, 0xfb, 0x00, 0xda, 0xb2, 0x00, 0x37, 0x4f, 0x1e, 0x36, 0x00, 0x05, 0x00, 0x05, 0x00, 0x03, 0x00, 0x06, 0x00,
```

Comment:
[Ditto](https://github.com/wazuh/wazuh/pull/13980/files#r920349447)

Reply:
Got it